newrelic/hooks/external_botocore.py

```diff
@@ -951,6 +970,10 @@ def _nr_endpoint_make_request_(wrapped, instance, args, kwargs):
     with ExternalTrace(library="botocore", url=url, method=method, source=wrapped) as trace:
         try:
             trace._add_agent_attribute("aws.operation", operation_model.name)
+            lambda_arn = getattr(instance._event_emitter, "_nr_lambda_arn", None)
+            if lambda_arn:
+                trace._add_agent_attribute("cloud.platform", "aws_lambda")
+                trace._add_agent_attribute("cloud.resource_id", lambda_arn)
```

Comment:
Are we confident this object isn't reused for future requests?

Reply:
I've updated this to use a different object that is generated for each request.